### PR TITLE
256 upgrade cx oracle to python oracledb

### PR DESCRIPTION
**Pull request by vd2org**

### CODE REVIEW

File: core/src/datayoga_core/__init__.py [hunk 1]
```diff
@@ -1,6 +1,23 @@
 import logging
 from typing import Any, Dict, List, Optional
 
+try:
+    # This hack makes it possible to use the new driver with SQLAlhemy 1.4.*
```
Comment (spicy-sauce) [inline core/src/datayoga_core/__init__.py]: typo: SQLAlhemy --> SQLAlchemy

File: core/src/datayoga_core/__init__.py [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+    from oracledb.exceptions import DatabaseError
+
+    oracledb.version = "8.3.0"
+    sys.modules["cx_Oracle"] = oracledb
+    try:
+        oracledb.init_oracle_client()
+    except DatabaseError:
+        pass
+except ImportError:
+    pass
```
Comment (spicy-sauce) [inline core/src/datayoga_core/__init__.py]: can't we have this snippet as a util so we won't have code duplication and use it here and in the test?

**Review by spicy-sauce — DISMISSED** [review]

File: core/src/datayoga_core/__init__.py (hunk 1 above)
Comment (spicy-sauce) [reply core/src/datayoga_core/__init__.py]: fixed

File: core/src/datayoga_core/__init__.py [hunk 3]
```diff
@@ -1,6 +1,24 @@
 import logging
 from typing import Any, Dict, List, Optional
 
+try:
```
Comment (zalmane) [inline core/src/datayoga_core/__init__.py]: @vd2org This should belong in the write.relational block. the main datayoga.core should be agnostic to what the blocks do.

Comment (zalmane) [reply core/src/datayoga_core/__init__.py]: and the snippet can be moved to the utils in relational block folder (so can be reused with the test)

File: core/src/datayoga_core/__init__.py (hunk 2 above)
Comment (vd2org) [reply core/src/datayoga_core/__init__.py]: No. This code should be executed before other imports. It's necessary.

File: core/src/datayoga_core/__init__.py (hunk 3 above)
Comment (vd2org) [reply core/src/datayoga_core/__init__.py]: The same as above.

Comment (zalmane) [reply core/src/datayoga_core/__init__.py]: why? I would assume it just needs to happen before the oracle import happens the first time. which can happen in the block init. can you explain?

File: core/src/datayoga_core/__init__.py (hunk 2 above)
Comment (spicy-sauce) [reply core/src/datayoga_core/__init__.py]: I understand, but why can't be a util? but this is the same issue raise by @zalmane, so please refer it there.

**Review by zalmane — APPROVED** [review]

**Review by zalmane — APPROVED** [review]

**Pull request merged**